### PR TITLE
Add bin/qt script

### PR DESCRIPTION
**Pull request by seanh**
A script for quickly running the code formatters, linters, tests, etc on
just the files that've changed on your branch vs master.

Currently only works for LMS, and for backend only.

Usage (from your lms directory):

    ../dev/bin/qt

### CODE REVIEW

File: bin/qt [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+        cmd = fmt.format(env=env, cmd=cmd, args=" ".join(args))
+
+        if self._verbose:
+            print(cmd)
+
+        subprocess.run(cmd, shell=True, check=True)
+
+
+def main():
```
Comment (jon-betts) [inline bin/qt:68]: :+1: This should make this trivial to install as a command line option from a pip package

File: bin/qt [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        if src_files:
+            run("lint", "pylint", src_files)
+            run("lint", "pycodestyle", src_files)
+            run("lint", "pydocstyle", src_files)
+
+        if all_test_files:
+            run("lint", "pylint", all_test_files, "--rcfile=tests/.pylintrc")
+
+        if test_files:
+            run("tests", "pytest", test_files)
```
Comment (jon-betts) [inline bin/qt:104]:
```suggestion
            run("tests", "pytest", test_files, "-v")
```

Adding verbose would list the files being tests and match our `make test` output.

File: bin/qt [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+        if functest_files:
+            run("functests", "pytest", functest_files)
```
Comment (jon-betts) [inline bin/qt:107]:
```suggestion
            run("functests", "pytest", functest_files, "-v")
```

**Comment by seanh** [comment]
Let's send a separate pr to add a `-v` option to `qt` that causes it to run all the tools verbosely. I want the default behaviour of `qt` to be to run fast and produce little outpu

**Pull request merged**